coalib/misc/Caching.py

```diff
@@ -0,0 +1,81 @@
+import calendar
+import time
+import hashlib
+
+from coalib.misc.CachingUtilities import (
+    pickle_load, pickle_dump, time_consistent, update_time_db)
+
+
+class FileCache:
+
+    def __init__(self, log_printer, project_dir, flush_cache=False):
+        """
+        Initialize cache instance for project.
+
+        :param log_printer: A LogPrinter object to use for logging.
+        :param project_dir: The root directory of the project to be used
+                            as a key identifier.
+        :param flush_cache: Flush the cache and rebuild it.
+        """
+        self.log_printer = log_printer
+        self.project_dir = project_dir.encode("utf-8")
+        self.md5sum = hashlib.md5(self.project_dir).hexdigest()
+        self.data = {}
+        if not time_consistent(log_printer, self.md5sum):
+            log_printer.warn("It seems like you went back in time - your "
+                             "system time is behind the last recorded run "
+                             "time on this project. The cache will "
+                             "be flushed and rebuilt.")
+            flush_cache = True
+        if not flush_cache:
+            self.data = pickle_load(log_printer, self.md5sum, {})
+        else:
```

Comment:
The file is anyway going to get rewritten at the end of execution. But yeah, deleting it might give the user a sense of satisfaction (that his new cache will be completely independent and clean), I guess.